### ISSUE TITLE
[Lisp] `make` error: bad type specifier

### ISSUE THREAD

**Issue by spinnerwheel** [spinnerwheel/object-oriented#6]
While trying to eval
```lisp
(def-class 'person nil (fields (name "default" string)))
; PERSON

(make 'person 'name "newname")
; bad thing to be a type specifier: ((SIMPLE-ARRAY CHARACTER (7))
```

**Comment by spinnerwheel** [comment]
The problem reside in the call to `subtypep-list-check` at line **303**
https://github.com/spinnerwheel/object-oriented/blob/8f59d63fd0ba6249d083275bfb22d065845ae0ef/Lisp/ool.lisp#L302-L317

With the second argument created with the `mapcar` call al line **313**

In the end, the call looks like this:
```prolog
(SUBTYPEP-LIST-CHECK (STRING) ((SIMPLE-ARRAY CHARACTER (7))))
```

Which leads to
```prolog
(SUBTYPEP ((SIMPLE-ARRAY CHARACTER (7))) (STRING))
```

And to the above error

*Note*: `((SIMPLE-ARRAY CHARACTER (7)))` is a list of lists

**Comment by DavideNicoletta** [comment]
Fixed with commit 6e57d79afd594d3898264044048cd822bcc42bcb
